src/cgutils.cpp

```diff
@@ -901,7 +901,7 @@ static LoadInst *emit_nthptr_recast(jl_codectx_t &ctx, Value *v, ssize_t n, MDNo
         emit_bitcast(ctx, vptr, PointerType::get(type, 0)))));
  }
 
-static Value *boxed(jl_codectx_t &ctx, const jl_cgval_t &v);
+static Value *boxed(jl_codectx_t &ctx, const jl_cgval_t &v,  bool is_promotable=false);
```

Comment:
So promotion is now always forbidden implicitly?

Reply:
Yes, the regression we were seeing was from something getting promoted when we called `boxed` on a separate part fo the of the compilation and then hoist something that shouldn't be hoisted. 
In the case of the regression we were hosting a string allocation from inside an error to the main path of the function which then impeded an optimization.
From what I understand we should only hoist the allocation when we have nested structs and it's fields are promotable which we only check inside `emit_new_struct`